### ISSUE TITLE
[cli][yarn 2] Running as a package.json script throws undeclared dependency error

### ISSUE THREAD

**Issue by hydrosquall** [contiamo/restful-react#230]
**Describe the bug**

Yarn 2 is unable to find `tslib`, it looks like it's hoping to see dependencies declared somewhere else (maybe tslib needs to be a top-level key in `yarn.lock`.

**To Reproduce**
1. Install yarn 2
2. Run `restful-react` in a NPM package.json script, e.g.

```js
// package.json
{
"scripts": {
       "open-api-gen": "restful-react import --config restful-react.config.js",
      }
}
```

```bash
yarn run open-api-gen
```

**Expected behavior**

- Same result as if script is run with `npm run open-api-gen` or `yarn run open-api-gen` with yarn 1, which completes without error

**Screenshots**

```
code: 'MODULE_NOT_FOUND',
  pnpCode: 'UNDECLARED_DEPENDENCY',
  data: {
    request: 'tslib',
    issuer: '/Users/cameron.yick/PROJECT_NAME/.yarn/$$virtual/restful-react-virtual-54bfb579bb/0/cache/restful-react-npm-9.10.0-ccfb1034aa-1.zip/node_modules/restful-react/dist/bin/',
    issuerLocator: {
      name: 'restful-react',
      reference: 'virtual:38b383702932cf51ac616965f70525625e0d3edc443238c6d02e51402d3bfa04b2dea1c6911efa4662701c6e315fd8eb4de4c4ab354ad6432ff1395c90d19c0e#npm:9.10.0'
    },
    dependencyName: 'tslib',
    candidates: [
      'restful-react',
      'case',
      'chalk',
      'commander',
      'ibm-openapi-validator',
      'inquirer',
      'lodash',
      'lodash-es',
      'openapi3-ts',
      'qs',
      'react',
      'react-fast-compare',
      'request',
      'swagger2openapi',
      'url',
      'yamljs'
    ]
  },
  requireStack: [
    '/Users/cameron.yick/PROJECT_NAME/.yarn/$$virtual/restful-react-virtual-54bfb579bb/0/cache/restful-react-npm-9.10.0-ccfb1034aa-1.zip/node_modules/restful-react/dist/bin/restful-react-import.js'
  ]
}
```

**Additional context**

- This is not an urgent issue for me, but raising it since others may encounter it as they move from yarn 1 to yarn 2 as well.

**Comment by arcanis** [comment]
> Yarn 2 is unable to find `tslib`, it looks like it's hoping to see dependencies declared somewhere else (maybe tslib needs to be a top-level key in `yarn.lock`.

No, it's just that `restful-react` declares `tslib` in [`devDependencies` only](https://cdn.jsdelivr.net/npm/restful-react@9.10.0/package.json), but `devDependencies` are ignored when the package is installed from the registries.

For the problem to be fixed, `restful-react` needs to list `tslib` as either a `dependencies` or `peerDependencies` entry - the first one looks more what I'd expect a library to use (since `tslib` is an helper library there's little reason to allow the user to provide its own version), but the second one is more in line with what you already have (since you already require your user to provide their own version somehow).

**Comment by fabien0102** [comment]
Nice catch, let's move tslib to `dependencies`!

**Comment by fabien0102** [comment]
After a bit more investigation, tslib is not even import/require anywhere… so… I guess we can just remove it 😁 I will try with yarn 2 to see if this resolve the issue

**Comment by fabien0102** [comment]
Fixed in `v9.10.1` 🎉
